examples/actix-web.rs

```diff
@@ -0,0 +1,103 @@
+use std::sync::Mutex;
+
+use actix_web::{web, App, HttpResponse, HttpServer, Responder, Result};
+use prometheus_client::encoding::text::{encode, Encode};
+use prometheus_client::metrics::counter::Counter;
+use prometheus_client::metrics::family::Family;
+use prometheus_client::registry::Registry;
+
+#[derive(Clone, Hash, PartialEq, Eq, Encode)]
+pub enum RequestType {
+    Get,
+    Post,
+}
+
+#[derive(Clone, Hash, PartialEq, Eq, Encode)]
+pub struct RequestTypeLabels {
+    pub request_type: RequestType,
+}
+
+/// Holds all metrics.
+/// We shouldn't store metrics inside a MetricsCollector -
+/// overwise we would have to take a lock for each increment of any metric
+pub struct Metrics {
+    requests: Family<RequestTypeLabels, Counter>,
+}
+
+impl Metrics {
+    pub fn new() -> Self {
+        Self {
+            requests: Family::default(),
+        }
+    }
+
+    pub fn inc_requests(&self, request_type: RequestType) {
+        self.requests
+            .get_or_create(&RequestTypeLabels {
+                request_type: request_type,
+            })
+            .inc();
+    }
+}
+
+/// Registers and collects metrics
+pub struct MetricsCollector {
```

Comment:
Why this wrapper around `Registry`? Despite the fact that `client_rust` does use the `Collector` abstraction (yet see https://github.com/prometheus/client_rust/issues/49), if I am not mistaken this concept is not wrapping a `Registry` in the other implementations, but rather the other way around.

Reply:
I tried to avoid Mutex here in this abstraction, but didn't found the one.. would like to see any suggestions

Reply:
With https://github.com/prometheus/client_rust/issues/57 fixed, is this still needed @oriontvv?

Reply:
With https://github.com/prometheus/client_rust/issues/57 fixed, is this still needed @oriontvv?

Reply:
Thanks, it looks like the PR fixes this issue, will try to adapt the example in few days.